RecoEgamma/EgammaElectronProducers/python/gsfElectrons_cfi.py

```diff
@@ -118,8 +118,8 @@
     #severityLevelCut = cms.int32(4),
 
     # Isolation algos configuration 
-    trkIsol03Cfg = trkIsol03CfgV2,
-    trkIsol04Cfg = trkIsol04CfgV2,                                        
+    trkIsol03Cfg = trkIsol03CfgV3,
+    trkIsol04Cfg = trkIsol04CfgV3,                                        
```

Comment:
just to make sure: is this change supposed to preserve the PF behavior?
I though that it was supposed to be V1.
I'd like to avoid changes in PF internals at this very late step in developments in 94X.
It would be OK for 95X.

Reply:
gsfElectrons are not used in PF, nor really anywhere in fact.

gedGsfElectrons are used. If this change happens, it has to have an extremely minor effect on particle flow electrons. On an individual level there may be minor edge case changes but overall there will be no effect. I understand your concern here but I think in this case, its needless.  If this breaks particle flow, particle flow is already broken.  

The comment here explains my reasoning:
https://github.com/cms-sw/cmssw/pull/20550#discussion_r140733786

Reply:
this change is for ecalDrivenGsfElectrons, which are an internal part of PF.

Reply:
after looking more carefully, while ecalDrivenGsfElectrons also enter the game in PFProducer, 
the tk isolation is used only for gedGsf

Reply:
So yes, you are correct that ecalDrivenGsfElectrons are internal in PF. I hadnt fully remembered that but yes it makes sense as theres a wee bit of a chicken and an egg scenario here. 

However as far as I'm aware the value of track isolation which used in PF comes from gedGsfElectronsTmp, not ecalDrivenGsfElectrons. The two should probably be synced as much as possible. 

https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/python/particleFlow_cfi.py#L86

which is the thing that is fed into 
https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/src/PFAlgo.cc#L660-L662

which is where the cut on track isolation is:
https://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/src/PFEGammaFilters.cc#L87-L110

Note, in CMSSW RECO, the only place that uses the track isolation is therehttp://cmslxr.fnal.gov/ident?_i=dr03TkSumPt
the rest are offline monitoring or electron identification.

Reply:
Also to clarify, and this is something I hadnt fully appreciated is that it also uses the track pt of ecalDrivenElectrons in the MVA classifier for PF. This is essentially the same problem. They really need to be synced,  I think it would be bad if they werent. This is also an extremely nasty duplication issue.

Reply:
ah I hadnt seen your comment when I replied. So in conclusion, we were both right, the same cut is applied on both ecalDrivenElectrons and gedGsfElectrons involving the track isolation.

Reply:
> after looking more carefully, while ecalDrivenGsfElectrons also enter the game in PFProducer, the tk isolation is used only for gedGsf

Apparently this was not correct:
ecalDriven electrons have the MVA value set only if they pass isolation cut
https://github.com/cms-sw/cmssw/blob/CMSSW_9_4_0_pre1/RecoEgamma/ElectronIdentification/plugins/ElectronIdMVABased.cc#L138
The thresholds are the same 10/10 in barrel/endcap as in the PFAlgo

Oddly enough, the comparisons for the last version of the PR (cffe799) do not show any differences in content of PF candidates.
IIUC the only functional change was this switch from v2 to v3 for the ecalDriven